### PR TITLE
Update and delete were realized for currency

### PR DESCRIPTION
**Pull request by fructusmortus**

### CODE REVIEW

File: currency/views.py [hunk 1]
```diff
@@ -19,10 +19,27 @@ def create_currency(request):
...
-def update_currency(request):
-    form = CurrencyForm()
+def update_currency(request, pk):
+    currency = Currency.objects.get(id=pk)
+    form = CurrencyForm(instance=currency)
+    if request.method == 'POST':
+        form = CurrencyForm(request.POST, request.FILES, instance=currency)
+        if form.is_valid():
+            currency = form.save(commit=False)
+            currency.user = request.user
```
Comment (nargiza-sarkulova) [inline currency/views.py]: Only the currency creator can update the currency so creator can't be anyone else here.

File: currency/views.py [hunk 2]
```diff
@@ -19,10 +19,27 @@ def create_currency(request):
...
+            currency.save()
+            return redirect('/currencies')
     context = {'form': form}
-    return render(request, 'currency/create_currency.html', context)
+    return render(request, 'currency/currency_form.html', context)
+
+
+def delete_currency(request, pk):
+    currency = Currency.objects.get(id=pk)
+    if request.method == 'POST':
```
Comment (nargiza-sarkulova) [inline currency/views.py:43]: Why `DELETE` method is not used?

**Comment by nargiza-sarkulova** [comment]
Please include screenshots in the PR description.

File: currency/views.py (hunk 2 above)
Comment (fructusmortus) [reply currency/views.py:43]: I've just followed the tutorial but after your question I've searched a bit and found that the allowed methods in forms are only GET and POST. [https://softwareengineering.stackexchange.com/questions/114156/why-are-there-are-no-put-and-delete-methods-on-html-forms](url)

File: currency/views.py (hunk 1 above)
Comment (fructusmortus) [reply currency/views.py]: Ok, will be fixed

**Comment by fructusmortus** [comment]
All currencies
image
Delete currency
image
Prefilled form to update currency
image

File: currency/templates/currency/all_currencies.html [hunk 3]
```diff
@@ -39,6 +41,14 @@ <h5>All currencies</h5>
                                 <td>{{currency.series}}</td>
                                 <td><img src="{{currency.thumbnail.url}}"></td>
                                 <td>{{currency.date_created}}</td>
+                                <td>
+                                    {% if user.is_authenticated %}
```
Comment (nargiza-sarkulova) [inline currency/templates/currency/all_currencies.html:45]: How does the page look like when user is not logged in? Seems like this check should be higher up.

File: currency/templates/currency/currency_form.html [hunk 4]
```diff
@@ -6,7 +6,7 @@
 <form action="" method="POST" enctype="multipart/form-data">
     {% csrf_token %}
     {{form}}
-    <input type="submit" value="Add" class="mt16 btn btn-primary left" />
+    <input type="submit" value="Submit" class="mt16 btn btn-primary left" />
```
Comment (nargiza-sarkulova) [inline currency/templates/currency/currency_form.html:9]:
```suggestion
    <input type="submit" value="Update" class="mt16 btn btn-primary left" />
```

File: currency/templates/currency/delete.html [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+<br>
+<div class="row">
+    <div class="col-md-6">
+        <div class="card card-body">
+            <p>Are you sure you want to delete "{{currency}}"?</p>
+            <form action="{% url 'delete_currency' currency.id %}" method="POST" class="form-group">
+                {% csrf_token %}
+                <a class="btn btn-warning" href="{% url 'currency' %}">Cancel</a>
+                <input class="btn btn-danger" type="submit" value="Confirm">
```
Comment (nargiza-sarkulova) [inline currency/templates/currency/delete.html]: It's better to be explicit
```suggestion
                <input class="btn btn-danger" type="submit" value="Delete">
```

**Review by nargiza-sarkulova — APPROVED** [review]

File: currency/templates/currency/currency_form.html (hunk 4 above)
Comment (fructusmortus) [reply currency/templates/currency/currency_form.html:9]: It is the same template which creates currency, so I've decided to use a more neutral "Submit" and renamed template as currency_form

Comment (nargiza-sarkulova) [reply currency/templates/currency/currency_form.html:9]: Ok, leave it as it is for now but you probably can detect which action user is doing and rename accordingly while rendering.

**Pull request merged**